sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1733,6 +1740,10 @@ void ProgramManager::addImage(sycl_device_binary RawImg,
     m_KernelIDs2BinImage.insert(std::make_pair(It->second, Img.get()));
     KernelIDs->push_back(It->second);
 
+    CompileTimeKernelInfoTy DefaultCompileTimeInfo{std::string_view(name)};
+    m_DeviceKernelInfoMap.try_emplace(std::string_view(name),
+                                      DefaultCompileTimeInfo);
```

Comment:
IIUC, `name` is a local variable and we are using its `string_view` as key to `m_DeviceKernelInfoMap`. Could this lead to dangling pointers in the map when local variable goes out of scope?

Reply:
`name` is a `const char*` taken from the device image here, so there won't be any problems unless the entry outlives the module that contains that image.

Now, there **is** a more general issue here if we, say, add images A and B coming from different libraries and containing the same kernel name, and then drop just the image we took the `string_view` from. But that's not introduced by this PR and I'm planning to address that soon.